process/releases.md

```diff
@@ -28,3 +28,18 @@ repos. This means, that release notes have to describe:
 - changes in build requirements
 - expected build issues and possible solutions
 - known issues with workaround
+
+## The release of the moulin-based project with self-reference
+If the project contains a moulin configuration (yaml file) with self-reference,
+we must break the recursion. We should refer to the repo
+using the release tag that will be created after the merge of the fixing commit.
+Here is a practical example.
+We have repo `meta-xt-super-product` and we have to make the release `v3.0.0`.
+So we have to use the commit release tag, that will appear after the merge:
+```
+    - type: git
+      url: https://github.com/xen-troops/meta-xt-super-product.git
+      rev: v3.0.0
+```
+Just after the merge of the fixing commit we have to create the release
+tag `v3.0.0` on that commit.
```

Comment:
Well, this is a tricky one as there will be moment in time where we will have invalid moulin file because the commit is merged by tag still does not exist.  CI will fail for sure. Any unlucky user who might want to build the project at that exact moment will get an error... 

This is not an issue if we are using branches instead of tags. But we need tags for releases. I can't come with better solution right now, so probably we'll have to left this as is.

Reply:
Reworked. Now we use release branches instead of non-existing tags.